src/vanilla/Templates/ServiceClientTemplate.cshtml

```diff
@@ -126,7 +132,15 @@ class @(Model.Name)(SDKClient):
             self@(Model.RequiredConstructorParameters)@(Model.IsCustomBaseUri ? "" : ", base_url=None")):
 @EmptyLine
         self.config = @(Model.Name)Configuration(@(Model.ConfigConstructorParameters))
-        super(@(Model.Name), self).__init__(@(Settings.AddCredentials? "self.config.credentials" : PythonConstants.None), self.config)
+        @if (Settings.AddCredentials && (bool)Settings.CustomSettings["UseAPIKey"])
+        {
+        <text>
+        self.config.credentials = ApiKeyCredentials(in_headers={
+            'Api-Key': api_key,
```

Comment:
Being that there is no official default value, I would prefer --use-api-key=Api-Key instead of a boolean

Reply:
Ah I see what you mean now. Wouldn't that make it so you had to provide the API key when generating the client code? I'd prefer to be able to proved the API key when instantiating the client, but I might be misunderstanding

Reply:
Sorry if I was unclear, I expect this `--use-api-key=FooHeader` to generate this:
```python
        self.config.credentials = ApiKeyCredentials(in_headers={
            'FooHeader': api_key,
```
Since the string `Api-Key` is not generic enough to be hard-coded with the option.
Does it clarify?

Reply:
Ah yes, excellent. It was also brought to my attention that we might be able to grab this information directly from a `schema.yaml`. Ours looks something like this: 

```
securityDefinitions:
  Bearer:
    type: apiKey
    name: Api-Key
    in: header
```
But I'm not C# enough yet to know how to access that information. Do you think that would be a better solution than a command line kwarg?